frontend/src/utilities/PrivateRoute.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+const PrivateRoute = ({
+  component: Component,
+  isAuthenticated,
+  isLoading,
+  ...rest
+}) => (
+  <Route
+    {...rest}
+    render={(props) => {
+      if (isLoading) {
+        return <h2>Loading...</h2>;
+      } else if (!isAuthenticated) {
+        return <Redirect to="/login" />;
+      } else {
+        return <Component {...props} />;
+      }
+    }}
+  />
+);
+
+const mapStateToProps = (state) => ({
+  isAuthenticated: state.auth.isAuthenticated,
+  isLoading: state.auth.isLoading,
+});
+
+export default connect(mapStateToProps)(PrivateRoute);
```

Comment:
instead of using connect, could you use the `useSelector` hook for fetching state

Reply:
Yes, I did it.